programs/util.c

```diff
@@ -326,6 +326,23 @@ int UTIL_prepareFileList(const char *dirName, char** bufStart, size_t* pos, char
 
 #endif /* #ifdef _WIN32 */
 
+/* Check if the file is precompressed (.zst, .lz4, .gz, .xz).
+YES => Skip the file (return 0)
+NO => return 1
+*/
+
+int UTIL_isCompressedFile(const char *inputName, const char *extensionList[])
+{
+   while(*extensionList!=NULL)
+   {
+     const char* ext = strstr(inputName,*extensionList);
```

Comment:
This will find the extension anywhere in the filename. E.g.: `foo.zst.baz.tar`. The extension check needs to check the end of the file.

Reply:
I do agree that the function does not explicitly check for file extension although it is not required for solving the issue#1579.
However, I have added a utility function in the latest commit that extracts extension from filename and compares only the extension with the list.